src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,731 @@
+     * @param operatorSetKey the encoded operatorSet to get queued allocation/deallocations for
+     * @dev Does not make an assumption on the number of pending allocations or deallocations at a time
+     */
+    function _getPendingAllocationsAndDeallocations(
+        address operator,
+        IStrategy strategy,
+        bytes32 operatorSetKey
+    ) internal view returns (uint256 numPendingAllocations, uint256 numPendingDeallocations) {
+        // Get pending allocations
+        Snapshots.History storage magnitudeUpdates = _magnitudeUpdate[operator][strategy][operatorSetKey];
+
+        // Read current magnitude's respective array index and length.
+        (,uint256 latestAllocPos, uint256 allocLength) =
+            magnitudeUpdates.upperLookupRecentWithPos(uint32(block.timestamp));
+
+        if (latestAllocPos < allocLength - 1) {
+            numPendingAllocations = allocLength - 1 - latestAllocPos;
+        }
+        // Else, latestAllocPos >= allocLength - 1;
+        // latestAllocPos cannot be greater than length - 1, thus when pos == length - 1 there are no pending allocations
+
+        // Get pending deallocations
+        uint256 deallocationsLength = _queuedDeallocationIndices[operator][strategy][operatorSetKey].length;
+        for (uint256 i = deallocationsLength; i > 0; --i) {
+            // index of pendingFreeMagnitude/deallocation to check for slashing
+            uint256 index = _queuedDeallocationIndices[operator][strategy][operatorSetKey][i - 1];
+
+            // If completableTimestamp is greater than completeUntilTimestamp, break
+            if (block.timestamp < _pendingFreeMagnitude[operator][strategy][index].completableTimestamp) {
+                ++numPendingDeallocations;
+            } else {
+                break;
+            }
+        }
+    }
+
+    /// @dev gets the latest total magnitude or overwrites it if it is not set
+    function _getLatestTotalMagnitude(address operator, IStrategy strategy) internal returns (uint64) {
+        (bool exists,, uint224 totalMagnitude) = _totalMagnitudeUpdate[operator][strategy].latestSnapshot();
+        if (!exists) {
+            totalMagnitude = WAD;
+            _totalMagnitudeUpdate[operator][strategy].push({key: uint32(block.timestamp), value: totalMagnitude});
+            operatorMagnitudeInfo[operator][strategy].freeMagnitude = WAD;
+        }
+
+        return uint64(totalMagnitude);
+    }
+
+    /// @dev gets the pending free magnitude available to add by completing numToComplete pending deallocations
+    /// and returns the next index to start from if completed.
+    function _getPendingFreeMagnitude(
+        address operator,
+        IStrategy strategy,
+        uint16 numToComplete,
+        uint192 nextIndex
+    ) internal view returns (uint64 freeMagnitudeToAdd, uint192 completed) {
+        uint256 pendingFreeMagnitudeLength = _pendingFreeMagnitude[operator][strategy].length;
+        freeMagnitudeToAdd = 0;
```

Comment:
This line isn't needed. The variable is automatically initialized to zero.

Reply:
Fixed in #786